packages/wdio-allure-reporter/src/utils.ts

```diff
@@ -113,6 +119,60 @@ export const getErrorFromFailedTest = (
     return test.error
 }
 
+/**
+ * Update the hook information with the new one, it could be use when a hook ends
+ * @param {HookStats} newHookStats - New information that will be applied to current hook info
+ * @param {ExecutableItemWrapper} hookElement - hook element registered in the report
+ * @param {AllureStep} hookRootStep - root hook step
+ *
+ * @private
+ */
+export const getHookStatus = (newHookStats: HookStats, hookElement: ExecutableItemWrapper, hookRootStep: AllureStep) => {
+    // stage to finish for all hook.
+    hookElement.stage = hookRootStep.stage =
+        Stage.FINISHED
+    // set error detail information
+    const formattedError = getErrorFromFailedTest(newHookStats)
+    hookElement.detailsMessage = hookRootStep.detailsMessage = formattedError?.message
+    hookElement.detailsTrace = hookRootStep.detailsTrace = formattedError?.stack
+
+    let finalStatus = Status.PASSED
+    // set status to hook root step
+    const hookSteps = hookRootStep.wrappedItem.steps
+    if (Array.isArray(hookSteps) && hookSteps.length) {
```

Comment:
@BorisOsipov use expect does not work fine in **All** hooks, why are you using it in that places? I saw that hooks does not recognize the status and errors using assertion in that places. I think that is a limitation with (All) fixtures.

The method **getTestStatus**, is only covering test cases scope. And the hook as fixtures has a different behavior, please check the object response.

Each Hooks are inside a test case scope so it will be affect the hook object status and error.

The behavior is expected in your test case, use throw error instead to see a similar behavior. I check this with a three different projects, cucumber, mocha, jasmine and mi company project (use mocha)

Reply:
Ok, @BorisOsipov AssertionError won't be recognized in 'all' type fixtures. This is because you can use it, but the hook object that leads to the methods is not affected in any way, and the error is not logged as it's not within the context of a test case yet. Previously, the service considered all fixtures as steps within the context of a test case or as a test case within the context of a suite, and your solution worked, but now it doesn't work that way.

Reply:
In the code below the hook will be broken, and the test case will be broken, because the **beforeAll** fixture is in the suite scope not test, and the test does not have a assertion error. So the test should be BROKEN.
```
describe('foo', () => {
    beforeAll(async () => {
        expect(true).toEqual(false)
    });
    it('t1', async () => {
    })
})
```

In the code below the test case will be failed, because the **beforeEach** fixture is in the test (t1) scope, and the test it will be affected by the fixture assertion. So the test should be FAILED. 
```
describe('foo', () => {
    beforeEach(async () => {
        expect(true).toEqual(false)
    });
    it('t1', async () => {
    })
})
```

It's the way are fixtures works not my way. I read all, and I debug the objects with different implementations, I saw allure-js-commons code, and you are asking me to change fixtures for test/steps again.

Reply:
@christian-bromann could you give us your opinion on this? I think that we are having a misunderstanding.

Reply:
@lararojasmr You seem to have more experience with the actual implementation details. Could you provide an example of test project code where the following `getHookStatus` simplification causes problems? I mean sample code and screen like I do in https://github.com/webdriverio/webdriverio/pull/11475#issuecomment-1791399930

```
export const getHookStatus = (newHookStats: HookStats, hookElement: ExecutableItemWrapper, hookRootStep: AllureStep) => {
    hookElement.stage = hookRootStep.stage = Stage.FINISHED
    const formattedError = getErrorFromFailedTest(newHookStats)
    hookElement.detailsMessage = hookRootStep.detailsMessage = formattedError?.message
    hookElement.detailsTrace = hookRootStep.detailsTrace = formattedError?.stack
    const finalStatus = getTestStatus(newHookStats)
    hookElement.status = hookRootStep.status = finalStatus
}
```

I find it extremely difficult to understand this problem without an example. Please explain with example.

Reply:
>I think that we are having a misunderstanding.

Yeap. Now I get want you want to explain. Let me think.